spatialmath/mesh.go

```diff
@@ -398,13 +373,16 @@ func (m *Mesh) ToPoints(density float64) []r3.Vector {
 	pointMap := make(map[string]r3.Vector)
 
 	// Add all triangle vertices, formatting as a string for map deduplication
-	for _, tri := range m.triangles {
+	for i, tri := range m.triangles {
+		centroid := r3.Vector{}
 		for _, pt := range tri.Points() {
 			// Transform point to world space
 			worldPt := Compose(m.pose, NewPoseFromPoint(pt)).Point()
+			centroid = centroid.Add(worldPt)
```

Comment:
this line and what "centroid" does confuses me.
Suppose a mesh made of a single triangle. `[(0, 0, 0), (1, 0, 0), (1, 1, 0)]` whose pose is the ZeroPose
centroid then becomes:
`(0, 0, 0) -> (1, 0, 0) -> (2, 1, 0)`
This seems incorrect?


I also have no clue what this line does and why is exists
```
centroid.Mul(1. / 3.)
```

Reply:
The centroid calculation of three points happens in this line:
```centroid.Mul(1. / 3.)```
The sum of the points is necessary in the loop to use the point helper function sum the vertices before calculating their centroid.

Ray, since this sufficiently confuses people, just call the variable on 381 `vecSum` or something and just have the centroid be returned through a `vecSum.Mul(1./3.)` calcualtion.

Reply:
Actually reading through this again this whole function is pretty fishy...